### PR TITLE
Ak customer controller

### LINKED ISSUE #2: Allow developers to access Customer resource
Verbs to be supported

1. GET
1. POST
1. PUT

* User should be able to GET a list of customers, and GET a single customer.
* If the query string parameter of `?_include=products` is provided, then any products that the customer is selling should be included in the response.
* If the query string parameter of `?_include=payments` is provided, then any payment types that the customer has used to pay for an order should be included in the response.
* If the query string parameter of `q` is provided when querying the list of customers, then any customer that has property value that matches the pattern should be returned.

If `/customers?q=Mic` is requested, then any customer whose first name is Michelle, or Michael, or Mick should be returned. Any customer whose last name is Michaelangelo, or Micacci, Micajkov should be returned.  Every property of the customer object should be checked for a match.

### PR DESCRIPTION
**Pull request by Akeen11**
## Link to Ticket
Closes #2 

## Description of Proposed Changes
- Created model for Customer
- Created mvc for Customer

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout <AK-CustomerController>
Add filler Customer data into your SQL
Go to Bangazon.sql folder, copy and paste initial data into database for testing
Test POST, PUT, DELETE, and GET in POSTMAN
```

## Impacted Areas in Application

List general components of the application that this PR will affect:

* CustomerModel and CustomerModelController

## Mentions @username

@davidnicholastaylor 
@KaylaReid 
@askingalot 

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.

### CODE REVIEW

**Review by KaylaReid — APPROVED** [review]
looks great

**Review by davidnicholastaylor — APPROVED** [review]
Looks great

File: Bangazon/Controllers/CustomersController.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                        }
+                        paymentReport[Customer.Id].PaymentTypes.Add(PaymentType);
+
+                        return Customer;
+                    }
+                );
+                return Ok(paymentReport.Values);
+            }
+
+            Console.WriteLine(sql);
```
Comment (askingalot) [inline Bangazon/Controllers/CustomersController.cs]: This was great for debugging, but itsn't needed now.

File: Bangazon/Controllers/CustomersController.cs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+using Bangazon.Models;
+
+//Aaron Keen
+//This controller allows you to get all customers, get a single customer, query the products the customer is selling, query the payment types the customer has used to pay, post a new customer, and edit an existing customer
+
+namespace Bangazon.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomersController : Controller
```
Comment (askingalot) [inline Bangazon/Controllers/CustomersController.cs]: Your controller should inherit from `ControllerBase`

File: Bangazon/Controllers/CustomersController.cs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        {
+            get
+            {
+                return new SqlConnection(_config.GetConnectionString("DefaultConnection"));
+            }
+        }
+
+        //GET api/Customers?q=Taco
+        [HttpGet]
+        public async Task<IActionResult> Get(string q, string _include)
```
Comment (askingalot) [inline Bangazon/Controllers/CustomersController.cs:39]: What does this method do if I pass both a `q` and and `_include` parameter?

For example, what happens if I use a URL such as http://localhost:5000/api/Customer?q=bob&_include=payments

**Review by askingalot — CHANGES_REQUESTED** [review]
I'd say this is 95% there. It's good work, but just needs one or two more things. See my other comments.

File: Bangazon/Controllers/CustomersController.cs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+            if (q != null)
+            {
+                string isQ = $@"
+                    AND c.FirstName LIKE '%{q}%'
+                    OR c.LastName LIKE '%{q}%'
+                ";
+                sql = $"{sql} {isQ}";
+            }
+
+            if (_include == "products")
```
Comment (askingalot) [inline Bangazon/Controllers/CustomersController.cs]: When _include=products is specified and a customer has not created any products, that customer is not being returned. I would expect to see the customer with an empty list of products.

**Review by askingalot — CHANGES_REQUESTED** [review]

**Review by askingalot — APPROVED** [review]

**Pull request merged**